drivers/mdio/mdio_gpio.c

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) 2023 Aleksandr Senin
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT zephyr_mdio_gpio
+
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/mdio.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(mdio_gpio, CONFIG_MDIO_LOG_LEVEL);
+
+#define MDIO_GPIO_READ_OP  0
+#define MDIO_GPIO_WRITE_OP 1
+#define MDIO_GPIO_MSB      0x80000000
+
+struct mdio_gpio_data {
+	struct k_sem sem;
+};
+
+struct mdio_gpio_config {
+	struct gpio_dt_spec mdc_gpio;
+	struct gpio_dt_spec mdio_gpio;
+};
+
+typedef const struct mdio_gpio_config *dev_cfg_ptr;
```

Comment:
Don't define alias.
Declare type name directly.
We don't prefer define alias to abbreviate in zephyr.